src/main/java/org/owasp/wrongsecrets/challenges/ChallengesController.java

```diff
@@ -63,18 +62,13 @@ public ChallengesController(ScoreCard scoreCard, List<ChallengeUI> challenges, R
         this.spoilingEnabled = spoilingEnabled;
     }
 
-    private boolean checkId(int id) {
+    private void checkValidChallengeNumber(int id) {
         // If the id is either negative or larger than the amount of challenges, return false.
         if (id < 0 || id >= challenges.size()) {
-            return false;
+            throw new ResponseStatusException(
+                HttpStatus.NOT_FOUND, "challenge not found"
+            );
         }
-        return true;
-    }
-
-    @GetMapping
-    @Operation(description = "Returns the given expalantion text for a challenge")
```

Comment:
funny, this is indeed never called... thanks! (just checked with coverage and saw it now :D )